### PR TITLE
add beaconchain node state syncing

### PR DESCRIPTION
**Pull request by chaosma**
beaconchain syncing for non-beaconchain nodes

### CODE REVIEW

File: core/types/gen_tx_json.go [hunk 1]
```diff
@@ -7,8 +7,8 @@ import (
 	"errors"
 	"math/big"
 
-	"github.com/ethereum/go-ethereum/common"
-	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/harmony-one/harmony/vendor/github.com/ethereum/go-ethereum/common"
```
Comment (daywednes) [inline core/types/gen_tx_json.go]: can you please confirm that we have to specify the code from vendor? @chaosma is the right import @astralblue ?

File: node/node.go [hunk 2]
```diff
@@ -104,6 +104,7 @@ type Node struct {
 	Consensus              *consensus.Consensus // Consensus object containing all Consensus related data (e.g. committee members, signatures, commits)
 	BlockChannel           chan *types.Block    // The channel to send newly proposed blocks
 	ConfirmedBlockChannel  chan *types.Block    // The channel to send confirmed blocks
+	BeaconBlockChannel     chan *types.Block    // The channel to send beacon blocks for non-beaconchain nodes
```
Comment (daywednes) [inline node/node.go]: why do we need this? We do not propose new beacon blocks for regular nodes

File: node/node.go [hunk 3]
```diff
@@ -123,15 +124,17 @@ type Node struct {
 	State      State      // State of the Node
 	stateMutex sync.Mutex // mutex for change node state
 
-	TxPool *core.TxPool
-	Worker *worker.Worker
+	TxPool       *core.TxPool
+	Worker       *worker.Worker
+	BeaconWorker *worker.Worker // worker for beacon chain
```
Comment (daywednes) [inline node/node.go]: I believe we don't need it

File: node/node.go (hunk 2 above)
Comment (chaosma) [reply node/node.go]: this is the channel to send the received block to beacon syncing program; it's not the same as the channel in consensus process

File: node/node.go (hunk 3 above)
Comment (chaosma) [reply node/node.go]: Let me check if we can use one worker for multiple chains

Comment (rlan35) [reply node/node.go]: I think it's good to use two workers separately.

File: core/types/gen_tx_json.go (hunk 1 above)
Comment (chaosma) [reply core/types/gen_tx_json.go]: fixed

**Pull request merged**